lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/auth/DBOAuthenticationDAOImpl.java

```diff
@@ -167,16 +166,16 @@ public boolean checkUserCredentials(long principalId, String passHash) {
 	@WriteTransaction
 	public boolean revalidateSessionTokenIfNeeded(long principalId) {
 		// Determine the last time the token was re-validate.
-		Long lastValidatedOn = jdbcTemplate.queryForObject(SELECT_AUTHENTICATED_ON_FOR_PRINCIPAL_ID, new SingleColumnRowMapper<Long>(), principalId);
+		Date lastValidatedOn = jdbcTemplate.queryForObject(SELECT_AUTHENTICATED_ON_FOR_PRINCIPAL_ID, new SingleColumnRowMapper<Date>(), principalId);
 		long now = clock.currentTimeMillis();
 		/*
 		 * Only revalidate a token if it is past its half-life.
 		 * See: PLFM-3202 & PLFM-3206
 		 */
-		if(lastValidatedOn + HALF_SESSION_EXPIRATION < now){
+		if(lastValidatedOn.getTime() + HALF_SESSION_EXPIRATION < now){
 			// The session token needs to be revaldiated.
 			userGroupDAO.touch(principalId);
-			jdbcTemplate.update(UPDATE_AUTHENTICATED_ON, clock.currentTimeMillis(), UUID.randomUUID().toString(), principalId);
+			jdbcTemplate.update(UPDATE_AUTHENTICATED_ON, clock.now(), principalId);
```

Comment:
While this seems to work (e.g. I debugged the integration test on the revalidateSessionTokenIfNeeded and it goes through the update) I was a bit puzzled: e.g. we are passing a long instead of a Date. Probably better to do it in the query itself?

```
private static final String UPDATE_AUTHENTICATED_ON = 
			"UPDATE "+TABLE_AUTHENTICATED_ON+
			" SET "+COL_AUTHENTICATED_ON_AUTHENTICATED_ON+"= NOW(), "+COL_AUTHENTICATED_ON_ETAG+"=UUID()"+
			" WHERE "+COL_AUTHENTICATED_ON_PRINCIPAL_ID+"= ?";
```

Reply:
I think we pass the value in to allow mocking the time for unit tests.